src/ui/pages/SwapForm/SwapForm.tsx

```diff
@@ -261,6 +281,25 @@ export function SwapFormComponent() {
     spendPosition,
   ]);
 
+  const { receiveAsset, spendAsset } = swapView;
+
+  const priceImpact = useMemo(() => {
```

Comment:
Let's extract this to the separate file

Reply:
Ok, I see, not critical

Reply:
yeah, I already extracted a separate function `calculatePriceImpact`, but wanted to keep all mentions of `primaryInput` here, in SwapForm